sagemaker_processing/feature_transformation_with_sagemaker_processing_Dask/feature_transformation_with_sagemaker_processing_dask.ipynb

```diff
@@ -0,0 +1,490 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Create the Dask preprocessing script."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile preprocess.py\n",
+    "from __future__ import print_function, unicode_literals\n",
+    "import argparse\n",
+    "import json\n",
+    "import logging\n",
+    "import os\n",
+    "import sys\n",
+    "import time\n",
+    "import warnings\n",
+    "import boto3\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from tornado import gen\n",
+    "import dask.dataframe as dd\n",
+    "import joblib\n",
+    "from dask.distributed import Client\n",
+    "from sklearn.compose import make_column_transformer\n",
+    "from sklearn.exceptions import DataConversionWarning\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.preprocessing import (\n",
+    "    KBinsDiscretizer,\n",
+    "    LabelBinarizer,\n",
+    "    OneHotEncoder,\n",
+    "    PolynomialFeatures,\n",
+    "    StandardScaler,\n",
+    ")\n",
+    "\n",
+    "warnings.filterwarnings(action=\"ignore\", category=DataConversionWarning)\n",
+    "s3_client = boto3.resource(\"s3\")\n",
+    "attempts_counter = 3\n",
+    "attempts = 0\n",
+    "\n",
+    "\n",
+    "def upload_objects(bucket, prefix, local_path):\n",
+    "    try:\n",
+    "        bucket_name = bucket  # s3 bucket name\n",
+    "        root_path = local_path  # local folder for upload\n",
+    "\n",
+    "        s3_bucket = s3_client.Bucket(bucket_name)\n",
+    "\n",
+    "        for path, subdirs, files in os.walk(root_path):\n",
+    "            for file in files:\n",
+    "                s3_bucket.upload_file(\n",
+    "                    os.path.join(path, file), \"{}/output/{}\".format(prefix, file)\n",
+    "                )\n",
+    "    except Exception as err:\n",
+    "        logging.exception(err)\n",
+    "\n",
+    "\n",
+    "def print_shape(df):\n",
+    "    negative_examples, positive_examples = np.bincount(df[\"income\"])\n",
+    "    print(\n",
+    "        \"Data shape: {}, {} positive examples, {} negative examples\".format(\n",
+    "            df.shape, positive_examples, negative_examples\n",
+    "        )\n",
+    "    )\n",
+    "\n",
+    "\n",
+    "if __name__ == \"__main__\":\n",
+    "    parser = argparse.ArgumentParser()\n",
+    "    parser.add_argument(\"--train-test-split-ratio\", type=float, default=0.3)\n",
+    "    args, _ = parser.parse_known_args()\n",
+    "\n",
+    "    # Get processor scrip arguments\n",
+    "    args_iter = iter(sys.argv[1:])\n",
+    "    script_args = dict(zip(args_iter, args_iter))\n",
+    "    scheduler_ip = sys.argv[-1]\n",
+    "\n",
+    "    # Start the Dask cluster client\n",
+    "    try:\n",
+    "        client = Client(\"tcp://{ip}:8786\".format(ip=scheduler_ip))\n",
```

Comment:
Curious... does the Client constructor accept hostnames? Could always pass "algo-1" here as the scheduler hostname and avoid some wiring.

Reply:
@apacker Yes, the client can accept hostnames. However, when I tested it out with hostname, I was getting random errors from time to time about name resolutions and connection timeouts which were solved by using the IP of the first (master/scheduler) host.